src/liquid/components/ld-badge/ld-badge.tsx

```diff
@@ -0,0 +1,77 @@
+import { Component, Element, h, Host, Prop, State, Watch } from '@stencil/core'
+import { HTMLStencilElement } from '@stencil/core/internal'
+import { getClassNames } from 'src/liquid/utils/getClassNames'
+
+/**
+ * @virtualProp ref - reference to component
+ * @virtualProp {string | number} key - for tracking the node's identity when working with lists
+ */
+@Component({
+  assetsDirs: ['assets'],
+  tag: 'ld-badge',
+  styleUrl: 'ld-badge.css',
+  shadow: true,
+})
+export class LdBadge {
+  @Element() el: HTMLStencilElement
+
+  /** Defines badge custom color */
+  @Prop() brandColor?: boolean
+
+  /** Icon name. */
+  @Prop() icon?: string
+
+  /** The size of the badge, translated in rem */
+  @Prop() size?: 'sm' | 'lg'
+
+  @State() hasCustomIcon = false
+  @State() hasText = false
+
+  @Watch('size')
+  private updateIconSize() {
+    this.el.querySelectorAll('ld-icon').forEach((icon) => {
+      if (this.size !== undefined) {
+        icon.setAttribute('size', this.size)
+      } else {
+        icon.removeAttribute('size')
+      }
+    })
+    this.el.querySelectorAll('.ld-icon').forEach((icon) => {
+      if (this.size === 'sm') {
+        icon.classList.remove('ld-icon--lg')
+        icon.classList.add('ld-icon--sm')
+      } else if (this.size === 'lg') {
+        icon.classList.remove('ld-icon--sm')
+        icon.classList.add('ld-icon--lg')
+      } else {
+        icon.classList.remove('ld-icon--sm', 'ld-icon--lg')
+      }
+    })
+  }
+
+  componentWillLoad() {
+    this.hasCustomIcon = !!this.el.querySelector('[slot="icon"]')
+    this.hasText = !!this.el.textContent.trim()
+    this.updateIconSize()
+  }
```

Comment:
```suggestion

  componentWillLoad() {
    this.hasCustomIcon = !!this.el.querySelector('[slot="icon"]')
    this.hasText = !!this.el.textContent.trim()
  }
```

I'm a little confused here, as you explicitly overwrite `--ld-icon-size-md` in the CSS, but not `sm`/`lg`. I changed this so that the icon size is overwritten, no matter what size-prop is set. With these changes the `updateIconSize` method should no longer be required.